### DISCUSSION TITLE
making the tools idempotent.

### DISCUSSION THREAD

**Discussion by AHuusom** [actions/virtual-environments#3236] [Ideas]
I would really like if you could make all the tools idempotent.

This would be useful, since packer can retry after it failed.
So it would be nice, if Pester could test before and skip if the tool is already installed.

Today, if the whole process fails after a 5 hour run, it is frustrating having to start all over again, because it doesn't know how to continue from where it started even though the possibility to retry is present.

This could be fixed by making a check before running the install of the tools, maybe by using pester.

**Comment by dibir-magomedsaygitov** [comment]
@AHuusom Thank you for your report. We will take a look and share our thoughts on it.

**Comment by AlenaSviridenko** [comment]
Hi @AHuusom,
this topic more fits the discussion section, as it is something that will require significant process update if we decide to do that.
I am going to convert it into discussion, thanks for raising up this point.
